control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go

```diff
@@ -129,13 +129,14 @@ func ReconcileDeployment(deployment *appsv1.Deployment, ownerRef config.OwnerRef
 				Labels: cvoLabels(),
 			},
 			Spec: corev1.PodSpec{
-				AutomountServiceAccountToken: pointer.Bool(false),
+				ServiceAccountName:           manifests.ClusterVersionOperatorServiceAccount("").Name,
```

Comment:
this should only be set when needsManagementClusterMetricsAccess is true. We are on the way to e2e test all components not needing SA have AutomountServiceAccountToken: pointer.Bool(false),

Reply:
Fixed in the new commit https://github.com/openshift/hypershift/commit/ef85b6396926fce1b8443b74492f6450327b20ab.

The  `AutomountServiceAccountToken` and `ServiceAccountName` is now set only when `EnableCVOManagementClusterMetricsAccess` is `true`. Same applies to the creation of CVO's service account, role and role binding.